api_test.go

```diff
@@ -552,6 +553,86 @@ func TestGetOAuthClients(t *testing.T) {
 	ApiSpecRegister = nil
 }
 
+func TestResetHandler(t *testing.T) {
+	uri := "/tyk/reload/"
+
+	ApiSpecRegister = make(map[string]*APISpec)
+
+	makeSampleAPI(t, apiTestDef)
+
+	recorder := httptest.NewRecorder()
+	params := make(url.Values)
+
+	req, err := http.NewRequest("GET", uri+params.Encode(), nil)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	resetHandler(recorder, req)
+
+	if recorder.Code != 200 {
+		t.Fatal("Hot reload failed, response code was: ", recorder.Code)
+	}
+
+	if len(ApiSpecRegister) == 0 {
+		t.Fatal("Hot reload was triggered but no APIs were found.")
+	}
+}
+
+func TestGroupResetHandler(t *testing.T) {
+	signalChan := make(chan bool)
+	cacheStore := RedisClusterStorageManager{}
+	cacheStore.Connect()
+
+	go func() {
+		cacheStore.StartPubSubHandler(RedisPubSubChannel, func(message redis.Message) {
+			notif := Notification{}
+			if err := json.Unmarshal(message.Data, &notif); err != nil {
+				t.Fatal("Unmarshalling message body failed, malformed: ", err)
+			}
+			if notif.Command == NoticeGroupReload {
+				signalChan <- true
+			} else {
+				signalChan <- false
+			}
+		})
+	}()
+
+	uri := "/tyk/reload/group"
+
+	ApiSpecRegister = make(map[string]*APISpec)
+
+	makeSampleAPI(t, apiTestDef)
+
+	recorder := httptest.NewRecorder()
+	params := make(url.Values)
+
+	req, err := http.NewRequest("GET", uri+params.Encode(), nil)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	groupResetHandler(recorder, req)
+
+	if recorder.Code != 200 {
+		t.Fatal("Hot reload (group) failed, response code was: ", recorder.Code)
+	}
+
+	if len(ApiSpecRegister) == 0 {
+		t.Fatal("Hot reload (group) was triggered but no APIs were found.")
+	}
+
+	for {
+		recvSignal := <-signalChan
+		if recvSignal {
+			break
+		}
+	}
```

Comment:
I've added a comment, this seems to be the first test to involve pubsub, so when writing tests in the future, this could make it easier to run multiple tests that produce/consume notifications.